src/dmd/semantic3.d

```diff
@@ -1352,8 +1318,11 @@ private struct FuncDeclSem3
     // The FuncDeclaration subject to Semantic analysis
     FuncDeclaration funcdecl;
 
+    // Scopes:
     // Scope of analysis
     Scope* sc;
+    // Scope created to avoid name collisions
+    Scope* sc2;
```

Comment:
Yes, the Idea it to fold most of the `visit` into the struct so that all the used local variables become members of the struct. i.e. this will be a common theme.